data_model/src/isi.rs

```diff
@@ -995,32 +1034,63 @@ pub mod error {
         }
 
         /// Type error
-        #[derive(Debug, Display, Clone, PartialEq, Eq, FromVariant)]
+        #[derive(
+            Debug,
+            Display,
+            Clone,
+            PartialEq,
+            Eq,
+            Hash,
+            FromVariant,
+            Serialize,
+            Deserialize,
+            Encode,
+            Decode,
+            IntoSchema,
+        )]
+        #[cfg_attr(feature = "std", derive(thiserror::Error))]
         #[ffi_type]
         pub enum TypeError {
             /// Asset type assertion error
             #[display(
-                fmt = "Asset Ids correspond to assets with different underlying types. {_0}"
+                fmt = "Asset Ids correspond to assets with different underlying types, {_0}"
             )]
-            AssetValueType(Mismatch<AssetValueType>),
+            AssetValueType(#[cfg_attr(feature = "std", source)] Mismatch<AssetValueType>),
             /// Parameter type assertion error
-            #[display(fmt = "Value passed to the parameter doesn't have the right type. {_0}")]
-            ParameterValueType(Mismatch<Value>),
+            #[display(fmt = "Value passed to the parameter doesn't have the right type, {_0}")]
+            ParameterValueType(#[cfg_attr(feature = "std", source)] Box<Mismatch<Value>>),
             /// Asset Id mismatch
-            #[display(fmt = "AssetDefinition Ids don't match. {_0}")]
-            AssetDefinitionId(Mismatch<<AssetDefinition as Identifiable>::Id>),
+            #[display(fmt = "AssetDefinition Ids don't match, {_0}")]
+            AssetDefinitionId(
+                #[cfg_attr(feature = "std", source)]
+                Box<Mismatch<<AssetDefinition as Identifiable>::Id>>,
```

Comment:
```suggestion
                Mismatch<<AssetDefinition as Identifiable>::Id>,
```

not sure, but I don't think it's needed? unless the linter is complaining

Reply:
Yeah, it's complaining about variant sized because I introduced `Box` for another variant as we discussed somwhere above